src/Len/Environment/Command/Stores/Urls/UrlTransformer.php

```diff
@@ -0,0 +1,380 @@
+     * Setter for the environment property.
+     *
+     * @param string $environment
+     * @return TransformCommand
+     * @throws \InvalidArgumentException when $environment is not one of
+     *   'dev' or 'staging'.
+     */
+    public function setEnvironment($environment)
+    {
+        static $allowedEnvironments = array('staging', 'dev');
+
+        if (!in_array($environment, $allowedEnvironments, true)) {
+            throw new \InvalidArgumentException(
+                'Destination environment must be one of: '
+                . implode(', ', $allowedEnvironments)
+                . "; Yet received: {$environment}"
+            );
+        }
+
+        $this->_environment = $environment;
+
+        return $this;
+    }
+
+    /**
+     * Clean the given domain part by stripping off invalid characters and
+     * trimming off surrounding domain delimiters (.).
+     *
+     * @param string $part
+     * @return string
+     */
+    protected function cleanDomainPart($part)
+    {
+        return trim(
+            preg_replace(
+                '/[^a-z0-9\-\_\.]/i',
+                '',
+                $part
+            ),
+            '.'
+        );
+    }
+
+    /**
+     * Create a domain from a given store, or a default domain if none is given.
+     *
+     * @param null|\Mage_Core_Model_Store $store
+     * @return string
+     */
+    public function createDomain(\Mage_Core_Model_Store $store = null)
+    {
+        $rv = implode(
+            '.',
+            [$this->getProject(), $this->getEnvironment(), $this->getDomain()]
+        );
+
+        if (isset($store) && $store->isAdmin() === false) {
+            $storeCode = $store->getCode();
+
+            if ($storeCode !== $store::DEFAULT_CODE) {
+                $rv = "{$storeCode}-{$rv}";
+            }
+        }
+
+        return $rv;
+    }
+
+    /**
+     * Transform a given store into a new base url.
+     *
+     * @param \Mage_Core_Model_Store $store
+     * @param bool $secure
+     * @return string
+     * @throws \InvalidArgumentException when $secure is not a boolean
+     */
+    public function transformBaseUrl(
+        \Mage_Core_Model_Store $store,
+        $secure = false
+    )
+    {
+        if (!is_bool($secure)) {
+            throw new \InvalidArgumentException(
+                'Invalid value for $secure supplied. Expected a boolean. Got: '
+                . gettype($secure)
+            );
+        }
+
+        $currentUrl = $store->getBaseUrl($store::URL_TYPE_LINK, $secure);
+        $path = parse_url($currentUrl, PHP_URL_PATH);
+        $scheme = parse_url($currentUrl, PHP_URL_SCHEME);
+        $domain = $this->createDomain($store);
+
+        return "{$scheme}://{$domain}{$path}";
+    }
+
+    /**
+     * Get a list of rewrites for the store base URLs.
+     *
+     * @param \Mage_Core_Model_Store $store
+     * @return array
+     */
+    public function getStoreRewrites(\Mage_Core_Model_Store $store)
+    {
+        return [
```

Comment:
Short array notation is for PHP >= PHP 5.4.
Either I fix the notation, or we are realistic about it and change the minimal PHP version for this repo to 5.5, to reflect our own needs.